charts/monitoring/prometheus/rules/src/general/kube-apiserver.yaml

```diff
@@ -46,36 +46,38 @@ groups:
-        for {{ $labels.verb }} {{ $labels.resource }}.
-      runbook_url: https://docs.kubermatic.com/kubermatic/master/monitoring/runbook/#alert-kubeapilatencyhigh
-    expr: cluster_quantile:apiserver_request_duration_seconds:histogram_quantile{job="apiserver",quantile="0.99",subresource!="log",verb!~"^(?:LIST|WATCH|WATCHLIST|PROXY|CONNECT)$"} > 1
+      message: API server is returning errors for {{ $value }}% of requests.
+      runbook_url: https://docs.kubermatic.com/kubermatic/master/monitoring/runbook/#alert-kubeapierrorshigh
+    expr: |
+      sum(rate(apiserver_request_total{job="apiserver",code=~"^(?:5..)$"}[5m])) without(instance, pod)
+        /
+      sum(rate(apiserver_request_total{job="apiserver"}[5m])) without(instance, pod) * 100 > 10
     for: 10m
     labels:
-      severity: warning
+      severity: critical
 
-  - alert: KubeAPILatencyHigh
+  - alert: KubeAPITerminatedRequests
     annotations:
-      message:
-        The API server has a 99th percentile latency of {{ $value }} seconds
-        for {{ $labels.verb }} {{ $labels.resource }}.
-      runbook_url: https://docs.kubermatic.com/kubermatic/master/monitoring/runbook/#alert-kubeapilatencyhigh
-    expr: cluster_quantile:apiserver_request_duration_seconds:histogram_quantile{job="apiserver",quantile="0.99",subresource!="log",verb!~"^(?:LIST|WATCH|WATCHLIST|PROXY|CONNECT)$"} > 4
-    for: 10m
+      message: The kubernetes apiserver has terminated {{ $value | humanizePercentage }} of its incoming requests.
+      runbook_url: https://docs.kubermatic.com/kubermatic/master/monitoring/runbook/#alert-kubeapiterminatedrequests
+    expr: |
+      sum(rate(apiserver_request_terminations_total{%(kubeApiserverSelector)s}[10m]))
+        / 
+      (  sum(rate(apiserver_request_total{%(kubeApiserverSelector)s}[10m])) + sum(rate(apiserver_request_terminations_total{%(kubeApiserverSelector)s}[10m])) ) > 0.20
+    for: 5m
     labels:
-      severity: critical
+      severity: warning
 
-  - alert: KubeAPIErrorsHigh
+  - alert: KubeAPITerminatedRequests
     annotations:
-      message: API server is returning errors for {{ $value }}% of requests.
-      runbook_url: https://docs.kubermatic.com/kubermatic/master/monitoring/runbook/#alert-kubeapierrorshigh
+      message: The kubernetes apiserver has terminated {{ $value | humanizePercentage }} of its incoming requests.
+      runbook_url: https://docs.kubermatic.com/kubermatic/master/monitoring/runbook/#alert-kubeapiterminatedrequests
     expr: |
-      sum(rate(apiserver_request_total{job="apiserver",code=~"^(?:5..)$"}[5m])) without(instance, pod)
-        /
-      sum(rate(apiserver_request_total{job="apiserver"}[5m])) without(instance, pod) * 100 > 10
+      sum(rate(apiserver_request_terminations_total{%(kubeApiserverSelector)s}[10m]))
+        / 
+      (  sum(rate(apiserver_request_total{%(kubeApiserverSelector)s}[10m])) + sum(rate(apiserver_request_terminations_total{%(kubeApiserverSelector)s}[10m])) ) > 0.20
```

Comment:
`%(kubeApiserverSelector)` doesn't seem valid Prometheus query language to me. Is this maybe a jsonnet leftover?

Reply:
I missed it , Corrected it now!